src/mlagility/api/script_api.py

```diff
@@ -146,8 +148,27 @@ def benchmark_script(
     models_found: Dict[str, ModelInfo] = {}
 
     for script in input_scripts:
+        script_path, targets, encoded_input = decode_input_script(script)
+
+        # Filter scripts by label
+        if labels != []:
+            skip_script = False
+            user_labels = lab.to_dict(labels)
+            file_labels = lab.load_from_file(encoded_input)
+            for key in user_labels:
+                # Skip benchmarking if user_label is not in file_labels
+                if key not in file_labels:
+                    skip_script = True
+                    break
+                # A label key may point to multiple label values
+                # Skip if not all values of user_labels[key] are in file_labels[key]
+                elif not all(elem in user_labels[key] for elem in file_labels[key]):
+                    skip_script = True
+                    break
+            if skip_script:
+                continue
```

Comment:
This is pretty verbose because you're using a `continue` statement within a nested loop. 

Suggest either:
1. Move this logic outside of the `for script in input_scripts:` loop so that you can simply `remove()` unwanted scripts from `input_scripts` (in which case the filtering in #156 should do the same)
2. Rewrite this code block to remove the inner loop

Reply:
Since #156 just merged, please make sure the filtering method used here is stylistically consistent with the `--resume` filtering (feel free to refactor the `--resume` filtering if needed).

Reply:
I'm now checking whether `required_labels` is a subset of `script_labels` in a function, which makes this code a lot less verbose. Let me know what you think.